### DISCUSSION TITLE
How do I retrieve a list item and file with a Graph ItemID and DriveID?

### DISCUSSION THREAD

**Discussion by plamber** [pnp/pnpcore#1011] [Q&A] [answered]
Hello,
I have a document library referenced with the Graph API. I am working with drive items with their ID and Graph item IDs. I must find the SharePoint library, list item, and file referenced to the Drive ID and Graph Item ID.

Considering that I have the following information:
- SharePoint URL
- DriveId
- DriveItem Id
----------------------------------------------
- Which query would you perform to identify the library using the REST API uniquely?
- Is there a way to identify an item in the library having the DriveId and DriveItem Id using the REST API
- Do these query work even with large libraries?

I read that there are properties VroomItemId and VroomDriveId on the SharePoint end. But right now, I am unsure what the most efficient queries are on the PNP Core SDK level to get these items.

Thank you for your feedback,
Patrick

**Comment by jansenbe** [comment, marked as answer]
@plamber : you can get those values from the Graph id's, see https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Utilities/DriveHelper.cs

**Reply by plamber** [reply]
You are a life saver.
Thank you.

Do you think it might be beneficial to make the helper methods public instead of keeping them private/internal?
